common/src/parse.cpp

```diff
@@ -127,12 +123,10 @@ pcl::console::parse_argument (int argc, const char * const * argv, const char *
 int
 pcl::console::parse_argument (int argc, const char * const * argv, const char * str, unsigned int &val)
 {
-  int index = find_argument (argc, argv, str) + 1;
-
-  if (index > 0 && index < argc )
-    val = atoi (argv[index]);
-
-  return (index - 1);
+  int dummy;
+  const auto ret = parse_argument (argc, argv, str, dummy);
+  val = static_cast<unsigned int> (dummy);
+  return ret;
```

Comment:
The issue existed  before your changes, but `atoi` won't be able to cover the full numeric range of a `uint`.  `atol` should solve the issue.

Reply:
Now that you point it out, I see the UB there. I'll modify the code accordingly. Nice catch 😄